Revised the devdocs contribution guide (#4532)
magento/devdocs: .github/CONTRIBUTING.md

We welcome all types of contributions; from minor typo fixes to new topics.

DevDocs staff members and [Community Maintainers](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributors.html#/community-maintainers) review issues and pull requests on a regular basis. We do our best to address all issues as soon as possible, but working through the backlog takes time. We appreciate your patience.

## Rewards for contributions

DevDocs works with Magento Community Engineering teams and projects.
As you contribute PRs, you gain [Contribution Points](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#points).

If you write and contribute a full topic, we will add your name (or your company's name) at the top of the DevDocs page and link it to your blog or website!

## Get started

![Get started workflow](https://devdocs.magento.com/common/images/contribute-prerequisites.png)

1. Make sure you have a [GitHub account](https://github.com/signup/free).

    **Note for partners:** Add [2FA](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#two-factor) protection when contributing to Magento repositories.

> 1. Sign the [Magento Contributor Agreement](https://magento.com/content/magento-contributor-agreement).

Reviewer: One big point of feedback from first time contributions: It was confusing to have CLA signing in these steps. It was  better to have in the PR creation and submission steps. I can give you an updated image to match if you want to move it. I have the source for it.

Author: Thanks Lori. I removed it from this section. It already exists in the Create a PR section. No need for an image.

1. [Fork](https://help.github.com/articles/fork-a-repo/) the [DevDocs repository](https://github.com/magento/devdocs). Remember to [sync your fork](https://help.github.com/articles/syncing-a-fork/) and update branches as needed.
1. Review the [DevDocs guidelines](#contribution-guidelines).

## Contribute documentation

The following diagram shows the contribution workflow:

![Contributing workflow](https://devdocs.magento.com/common/images/contribute-write-submit-pr.png)

> **Tip!** If you are not sure where to start contributing, search for issues with the [`help wanted`](https://github.com/magento/devdocs/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) and [`good first issue`](https://github.com/magento/devdocs/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) labels. These issues receive higher priority for processing.

Reviewer: Would it be worth noting the groomed label issues have been reviewed by DevDocs and available for working?

Author: I don't think it's necessary to mention that we've reviewed them, but I'll add `groomed` to the list of labels.

You can contribute by creating an issue or pull request (PR) on our [DevDocs](https://github.com/magento/devdocs) GitHub repository.
We welcome all types of contributions; from minor typo fixes to new topics.

Magento's tech writer team and [Community Maintainers](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributors.html#/community-maintainers) review issues and pull requests on a regular basis. We do our best to address all issues as soon as possible, but working through the backlog takes time. We appreciate your patience.

## Rewards for contributions

DevDocs works with Magento Community Engineering teams and projects.
As you contribute PRs, you gain [Contribution Points](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#points).

If you write and contribute a full topic, we will add your name (or your company's name) at the top of the DevDocs page and link it to your blog or website!

## Get started

![Get started workflow](https://devdocs.magento.com/common/images/contribute-prerequisites.png)

> 1. Make sure you have a [GitHub account](https://github.com/signup/free) with [Two-Factor Authentication](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#two-factor)(2FA) enabled on your account. We require partners to add 2FA protection when contributing to Magento repositories.

Reviewer 2: This is a note for a particular audience and can be skipped by general audience, so it should be represented as a separate note for partners. For example: _**Note for partners**: Add 2FA protection when contributing to Magento repositories._
Also, do we have a description of a contributor/maintainer/partner/etc roles somewhere? How do I know what role should I be in when reading the guidelines?

Author: Good question. This is for contributors only.

I wrote some maintainer guidelines a few weeks ago after onboarding a couple of new maintainers. I shared with them directly, but I've been meaning to make them public. I sent @lorikrell a copy a few days ago so she could add to the repo wiki, but she's been at Imagine all week. Instead of adding them to the wiki, maybe we should add to the repo as `.github/MAINTAINER-GUIDELINES.md`.

Author: See https://github.com/magento/devdocs/pull/4535

1. Sign the [Magento Contributor Agreement](https://magento.com/content/magento-contributor-agreement).
1. [Fork](https://help.github.com/articles/fork-a-repo/) the [DevDocs repository](https://github.com/magento/devdocs). Remember to [sync your fork](https://help.github.com/articles/syncing-a-fork/) and update branches as needed.
1. Review the [DevDocs guidelines](#contribution-guidelines).

> ## Contribute docs

Reviewer 2: This is not part of the PR, but I'd like to highlight that we should avoid use of abbreviations: _docs_ -> _documentation_

Reviewer 3: I second this.

The following diagram shows the contribution workflow:

![Contributing workflow](https://devdocs.magento.com/common/images/contribute-write-submit-pr.png)

> **Tip!** If you are not sure where to start contributing, search for issues with the [`help wanted`](https://github.com/magento/devdocs/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) and [`good first issue`](https://github.com/magento/devdocs/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) labels.

Reviewer 4: We should also say that PR's linked to issues with the `help wanted` tag automatically have a high priority for processing.

### Create a branch

1. Create a new branch from your fork using a name that best describes the work or references a GitHub issue number.
1. Edit or create markdown (`.md`) files in your branch.
1. Push your branch to your fork.

### Create a pull request

1. Create a pull request to the [magento/devdocs](https://github.com/magento/devdocs) repository.

   In general, you should use `master` as the base branch when creating a PR. If your contribution is related to a release that is in progress, use a version-specific integration branch, such as `develop`.

1. Complete the pull request template.

>    We will close your pull request if you fail to complete the template.

Reviewer 4: ```suggestion
   **We will close your pull request if you fail to complete the template.**
```
Feel like this should be boldly stated.

Reviewer 3: Be ready to follow through on this. 🙂

> 1. After creating a pull request, a DevDocs staff member or maintainer will review it and may ask you to make revisions.

Reviewer 4: Should we add a note that a lack of response in a reasonable amount of time will result in a rejected PR?

Reviewer 2: There are two different names for the same entity in this topic:
1. Magento's tech writer team
2. DevDocs staff

It is not clear whether they are the same or different. We should use one name for one entity to avoid ambiguity.

**Note:** If you have not signed the [Magento Contributor Agreement](https://magento.com/content/magento-contributor-agreement), the pull request provides a link. Magento requires a signed form and agreement to the terms for contribution.

## Contribution guidelines

The following guidelines may answer most of your questions and help you get started:

> - Write content using markdown. See the [Templates](#templates) section for examples.

Reviewer 2: ```suggestion
- Write content using Markdown. See the [Templates](#templates) section for examples.
```

Reviewer 2: Our templates do not comply with pure Markdown markup. It makes sense to add information that we use [kramdown](https://kramdown.gettalong.org/) that implement several Markdown extensions.

Reviewer 2: The _Templates_ section contains links to [Markdown](https://github.com/magento/devdocs/blob/master/guides/v2.1/contributor-guide/templates/basic_template.md) files. I'm not sure that this is a good idea. For example, if I'll follow the provided guidelines how to add notes,
```
{%raw%}
{%
include note.html
type='info'
content='This is a note callout. You can use these to provide important information on a topic.'
%}
{%endraw%}
```

I won't see the expected note.

- Review existing [pull requests](https://github.com/magento/devdocs/pulls) and [issues](https://github.com/magento/devdocs/issues) to avoid duplicating work.
- For large contributions, or changes that include multiple files, [open an issue](#report-an-issue) and discuss it with us first. This helps prevent duplicate or unnecessary work.

> - Do not make global find/replace type changes without first creating and issue and discussing it with us. Global changes can have unintended consequences.

Reviewer 3: ```suggestion
- Do not make global find-and-replace changes without first creating an issue and discussing it with us. Global changes can have unintended consequences.
```

    - We use symbolic links for topics that are the same across versions of Magento. Remove the symbolic link and add a new file if your changes are unique to a specific version. Copy and paste a previous version of the topic to get started.

- Make sure that you update the correct version(s) of documentation (v2.1, v2.2, v2.3). If you are not sure what directory to put your content in, just do your best. We can help re-locate it (if necessary) during the review process.
- Review your work for basic typos, formatting errors, or ambiguous sentences before opening a pull request.
- Revise pull requests according to review feedback. We will close pull requests that require an inordinate amount of time to review and process (especially for minor changes) if you fail to make revisions according to review feedback.
- Do not directly contact DevDocs team members or maintainers on Slack to review your pull request unless it has been open for more than five days. We have a process and queue for pull requests that everyone must follow.

## Tips for writing content

Use the following guidelines to help you with the writing process:

> - Focus your efforts on providing useful information for your fellow Magento developers and community members. For example, new or revised code samples, important notes, clarification of vague or ambiguous content.

Reviewer 3: ```suggestion
- Focus your efforts on providing useful information for your fellow Magento developers and community members. For example, consider providing or revising code samples, important notes, and clarifying vague or ambiguous content.
```

- Define the goal of your topic. What exactly do you want to teach the reader?
- Make the title of your topic reflect the content.
- Keep your sentences concise. Separate conceptual information from procedural steps.
- Batch several small changes into a single pull request (instead of separate ones) to ensure your contributions are approved and merged quickly. Have several typo fixes across several areas of documentation? Combine them into a single PR.
- Remember to write in present tense, use the second person, and use active voice (not passive). For example, _"The log captures commands, output..."_.
- Use notes to alert readers about important details.
- Use cross-references to other topics sparingly. We can help you with the syntax if it is not clear. The template provides an example you can use.
- Need help with markdown? See our [templates](#templates).

### Review changes locally

> _(Optional)_ To review your changes in HTML output, follow the instructions in the [README](https://github.com/magento/devdocs/blob/master/README.md) to build the devdocs site locally using Jekyll.

Reviewer 2: Not sure if will work. The README does not contain any instructions about how to build and preview subrepositories.

Author: Contributors may not be able to build some repos even after we update the README since some repos are private.

### Templates

> We provide templates to help you get started writing new content and understanding markdown formatting:

Reviewer 2: ```suggestion
We provide templates to help you get started writing new content and understanding Markdown formatting:
```

| Property  | Description |
| ------------- | ---------- |
| `group`       | Defines the topic's guide or section. Use the table of contents `.yml` file name. This loads your left-side navigation. We will help during the PR process to add new files to the `.yml` file. |
| `title`       | Sets the title of the page in the HTML metadata and the main title on the page.  |

## Report an issue

If you find a typo or errors in Magento DevDocs, you can either fix it with a pull request (as described above) or you can report it by creating an issue in the DevDocs GitHub repository.

You must complete the issue template. We will close your issue if you fail to complete the template. Enter as much information as you can, including content corrections, steps to reproduce, command or code updates, or questions for clarifications.

**Note:** Check the existing [issues](https://github.com/magento/devdocs/issues) on GitHub to see if someone has already reported the issue.

> You have a couple of options to create an issue:

Reviewer 3: ```suggestion
You can provide feedback using one of the following options:
```
